src/share/poudriere/image.sh

```diff
@@ -262,11 +272,51 @@ usb|rawdisk)
 	EOF
 	;;
 *firmware)
+	# Configuring nanobsd-like mode
+	# It re-use diskless(8) framework but using a /cfg configuration partition
+	# It needs a "config save" script too, like the nanobsd example:
+	#  /usr/src/tools/tools/nanobsd/Files/root/save_cfg
+	# Or the BSDRP config script:
+	#  https://github.com/ocochard/BSDRP/blob/master/BSDRP/Files/usr/local/sbin/config
+	# Because rootfs is readonly, it create ramdisks for /etc and /var
+	# Then we need to replace /tmp by a symlink to /var/tmp
+	# For more information, read /etc/rc.initdiskless
 	cat >> ${WRKDIR}/world/etc/fstab <<-EOF
-	/dev/gpt/${IMAGENAME}0 / ufs ro 1 1
+	/dev/gpt/${IMAGENAME}1 / ufs ro 1 1
+	/dev/gpt/cfg /cfg ufs rw,noatime,noauto 2 2
+	/dev/gpt/data /data ufs rw,noauto,failok 2 2
 	EOF
-	mkdir -p ${WRKDIR}/world/conf/base
-	tar -C ${WRKDIR}/world -X ${excludelist} -cf - etc | tar -xf - -C ${WRKDIR}/world/conf/base
+	# Enable diskless(8) mode
+	touch ${WRKDIR}/world/etc/diskless
+	for d in cfg data; do
+		mkdir -p ${WRKDIR}/world/$d
+	done
+	# Declare system name into /etc/nanobsd.conf: Allow to re-use nanobsd script
+	echo "NANO_DRIVE=gpt/${IMAGENAME}" > ${WRKDIR}/world/etc/nanobsd.conf
+	# Move /usr/local/etc to /etc/local (Only /etc will be backuped)
+    if [ -d ${WRKDIR}/world/usr/local/etc ] ; then
+        mkdir -p ${WRKDIR}/world/etc/local
+		tar -C ${WRKDIR}/world -X ${excludelist} -cf - usr/local/etc/ | tar -xf - -C ${WRKDIR}/world/etc/local
+        rm -rf ${WRKDIR}/world/usr/local/etc
+        ln -s /etc/local ${WRKDIR}/world/usr/local/etc
+    fi
+
+	# Copy /etc and /var to /conf/base as "reference"
+	for d in var etc; do
+		mkdir -p ${WRKDIR}/world/conf/base/$d ${WRKDIR}/world/conf/default/$d
+		tar -C ${WRKDIR}/world -X ${excludelist} -cf - $d | tar -xf - -C ${WRKDIR}/world/conf/base
```

Comment:
Once poudriere-firmware boot, the resulting system show this kind of error message:
`sendmail: can not open config '/usr/local/etc/dma/dma.conf': Permission denied
Jul 14 12:00:00 router cron[2673]: _secure_path: cannot stat /etc/login.conf: Permission denied`

Then I've check the /etc permission:
```
[root@poudriere-img]~# ll / | grep etc
d---------  25 root  wheel     2560 Jul 13 18:23 etc/
```
/etc is a ram disk created by diskless(8) (like on nanobsd):
```
[root@poudriere-img]~# mount
/dev/gpt/BSDRP1 on / (ufs, local, read-only)
devfs on /dev (devfs, local, multilabel)
/dev/md0 on /etc (ufs, local)
/dev/md1 on /var (ufs, local)
```
On a standard FreeBSD and nanobsd permission of /etc should be:
```
[root@nanobsd]~# ll / | grep etc
drwxr-xr-x  27 root  wheel     2048 Jun 12 05:30 etc/
```


And if I unmount this /etc, the underlying /etc regain it's original (and good) permissions:
```
[root@poudriere-img]~# umount /etc
[root@poudriere-img]~# ll / | grep etc
drwxr-xr-x  24 root  wheel     2560 Jul 11 16:27 etc/
```

This is why I've no clue about where the permission problem is inserted during the poudriere image build.
About adding "-p" to the 2 tar commands: I've just tried it, but it didn't solve the problem.
I will be AFK since now to 8th August.